### ISSUE TITLE
[BUG] eXide hides Errors

### ISSUE THREAD

**Issue by adamretter** [eXist-db/eXide#260]
This query currently raises a ConcurrentModificationException in eXist-db:

```xquery
<parameters>{
    let $root := <a:b xmlns:a="ans" xmlns:c="cns"/>
    for $prefix in in-scope-prefixes($root)[not(. = ('xml', 'xsi'))]
      return
          namespace-uri-for-prefix($prefix,$root) !
              <param name="{$prefix}" value="{.}"/>
}</parameters>
```

However when it is run in eXide, it just returns 0 results with no indication that an error occurred. This is misleading.

To see the exception, run the query from the REST API or Java Admin Client.

**Comment by joewiz** [comment]
I can reproduce this on my system (eXist 5.3.0-SNAPSHOT + eXide 2.4.9). 

I believe the problem lies not in eXide per se but in eXide's use of the XQueryServlet for evaluating queries submitted by the user. See https://github.com/eXist-db/eXide/blob/develop/controller.xql#L164-L207.

If, instead of relying on the XQueryServlet to catch the error, we surround our own code with a try-catch block, it returns the expected error:

> `Q{http://exist.sourceforge.net/NS/exist/java-binding}java.util.ConcurrentModificationException`

Here's the code with the try-catch block:

```xquery
xquery version "3.1";

try { 
    <parameters>{
        let $root := <a:b xmlns:a="ans" xmlns:c="cns"/>
        for $prefix in in-scope-prefixes($root)[not(. = ('xml', 'xsi'))]
          return
              namespace-uri-for-prefix($prefix,$root) !
                  <param name="{$prefix}" value="{.}"/>
    }</parameters>
} catch * { 
    $err:code
}
```

**Comment by adamretter** [comment]
@joewiz How do you execute the code with the try-catch?

**Comment by joewiz** [comment]
@adamretter By selecting the "Eval" button.

> ![Screen Shot 2020-05-29 at 11 40 46 AM](https://user-images.githubusercontent.com/59118/83278297-5f5da280-a1a1-11ea-9b26-7da819222b16.png)

**Comment by adamretter** [comment]
@joewiz Okay, so through eXide still ;-)

**Comment by adamretter** [comment]
@joewiz I just executed the original query directly via the XQueryServlet. It seems that the XQueryServlet does return an error:

<img width="497" alt="Screenshot 2020-05-29 at 18 00 55" src="https://user-images.githubusercontent.com/1264057/83280120-7b2f6b80-a1d6-11ea-994b-eee553f0a020.png">

So are we are back to thinking the problem is indeed with eXide?

**Comment by joewiz** [comment]
@adamretter What is the contents of stuff.xql, and is any special configuration needed to call it at the URL in your screenshot? I’m not familiar with the /exist/xquery endpoint - forgive my ignorance.

**Comment by adamretter** [comment]
@joewiz The contents of `stuff.xql` are exactly the query at the top of this issue.

Inside an eXist-db distribution, I just stored it at `etc/webapp/xquery/stuff.xql`

**Comment by joewiz** [comment]
@adamretter I now understand why eXide is failing to report the error, which is because the error has no description. As your screenshot shows, the `<div class='description'>` where the error would normally appear simply says `null`. In eXide, this results in the error being hidden, since eXide expects error messages to contain a description. 

eXide gets such error descriptions directly from the XQueryServlet. As described in the URL Rewriting documentation, when [the `xquery.report-errors` attribute](https://exist-db.org/exist/apps/doc/urlrewrite#xq-servlet) is set to `yes` (as it is in the link I provided above),

> ... an error in the XQuery will not result in an HTTP error. Instead, the string message of the error is enclosed in an element `<error>` which is then written to the response stream. 

In other words, if there were any description for the error, eXide would show it. 

So here's the question for this issue:

Should (a) eXide be expected to catch errors with null descriptions? Or, should (b) eXist generally be supplying descriptions for errors it raises?

If the former is true and eXist should be expected to raise errors without descriptions, I have a fix working locally, in which I construct an error description in such cases. I've taken inspiration from what happens when you call `error()` but don't supply a QName or description, as described in https://www.w3.org/TR/xpath-functions-31/#func-error. Here's a screenshot:

> ![Screen Shot 2020-06-01 at 1 10 28 PM](https://user-images.githubusercontent.com/59118/83434907-54a54680-a409-11ea-8fba-74f17764062e.png)

**Comment by joewiz** [comment]
Answer from @adamretter on the Community Call: In general, errors should be expected to supply an error code and description, but it's possible eXist may raise an error that lacks these. 

--> So eXide should handle this scenario.
